src/pages/account.jsx

```diff
@@ -22,17 +22,24 @@ import { set } from "mongoose";
  */
 export default function Account() {
   const [editName, setEditName] = useState(false);
-  const [name, setName] = useState("");
+  const [name, setName] = useState("User");
+  const [role, setRole] = useState("User");
+  const [image, setImage] = useState(userpfpplaceholder);
   const [showDeactivateModal, setShowDeactivateModal] = useState(false);
   const { data: session, update } = useSession();
 
   const router = useRouter();
 
   useEffect(() => {
-    if (session) {
-      setName(session.user.name);
-    }
-  }, [session])
+    fetch(`/api/users/${session?.user._id}`)
```

Comment:
check if `session` and `session.user` != null/undefined before fetching. Another way to do it is check if `status` != "loading" and then add `status` to the dependency array. `status` comes from `useSession()` hook